llvm/lib/Target/ARM/ARMISelLowering.cpp

```diff
@@ -7966,7 +7953,7 @@ SDValue ARMTargetLowering::LowerBUILD_VECTOR(SDValue Op, SelectionDAG &DAG,
 
       if (Val.getNode()) {
         SDValue Vmov = DAG.getNode(ARMISD::VMOVIMM, dl, VmovVT, Val);
-        return DAG.getNode(ISD::BITCAST, dl, VT, Vmov);
+        return DAG.getNode(ARMISD::VECTOR_REG_CAST, dl, VT, Vmov);
       }
 
       // Try an immediate VMVN.
```

Comment:
Can you change the MVN below too (and maybe make sure there are tests to ensure it is correct).

Reply:
done

Reply:
> Can you change the MVN below too (and maybe make sure there are tests to ensure it is correct).

only change MVN, and all tests pass, [https://github.com/llvm/llvm-project/pull/105913](url)